src/rsz/src/Resizer.cc

```diff
@@ -2911,8 +2942,11 @@ int Resizer::holdBufferCount() const
 }
 
 ////////////////////////////////////////////////////////////////
-void Resizer::recoverPower(float recover_power_percent)
+void Resizer::recoverPower(float recover_power_percent,
+                           bool match_cell_footprint)
 {
+  utl::SetAndRestore<bool> set_match_footprint(match_cell_footprint_,
+                                               match_cell_footprint);
```

Comment:
nit: I think this SetAndRestore utility is actually harder to read than the equivalent tmp, set and restore stuff.

Reply:
It is more exception safe (and early return).

Reply:
Fair on both points. :)